src/main/java/net/snowflake/client/jdbc/SnowflakeFileTransferAgent.java

```diff
@@ -1016,6 +1016,19 @@ private void parseCommand() throws SnowflakeSQLException {
     }
   }
 
+  static StageInfo getStageInfo(JsonNode jsonNode, SFSession session) throws SnowflakeSQLException {
+
+    StageInfo stageInfo = getStageInfo(jsonNode);
+
+    // Update StageInfo to reflect use of S3 regional URL.
+    // This is required for connecting to S3 over privatelink when the
+    // target stage is in us-east-1
+    if (stageInfo.getStageType() == StageInfo.StageType.S3)
+      stageInfo.setUseS3RegionalUrl(session.getUseRegionalS3EndpointsForPresignedURL());
```

Comment:
Is "target-stage is in us-east-1' needed to be verified here? Other than that, should we have test cases for s3, non-s3, and s3-but-not-us-east-1?